elastic.go

```diff
@@ -137,6 +139,47 @@ func findIndices(es elastic.Client, indexGlob string) ([]string, error) {
 	return indices, err
 }
 
+func compressIndices(hits string) *bytes.Buffer {
```

Comment:
Can we use a io.Reader for return?

Reply:
`io.Reader` does not implement `Write()`, which is needed in order to feed to the compressor.

Reply:
If you need bot reader and writer there is `io.ReadWriter`

Reply:
`io.Pipe`?

Reply:
This returns a pointer to a buffer, which should be enough. If we see memory issues, we can take this in another PR.

Reply:
`bytes.Buffer` implements both `io.Reader` and `io.Writer`, so it's just the signature (and later usage of that) that would need changing (roughly no code in the function)

To me, using interfaces rather than fixed types in signatures feels very helpful, both for "regular usage" (e.g. chaining such as wrapping it in a StreamReader) and for testing.

Reply:
This is on the works on a different branch.